tiny-event-sourcing-lib/src/main/kotlin/ru/quipy/streams/EventStreamReadingStrategy.kt

```diff
@@ -0,0 +1,127 @@
+package ru.quipy.streams
+
+import kotlinx.coroutines.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import ru.quipy.domain.Aggregate
+import ru.quipy.domain.Event
+import ru.quipy.mapper.EventMapper
+import kotlin.reflect.KClass
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+interface EventStreamReadingStrategy<A : Aggregate> {
+    suspend fun read(stream: AggregateEventStream<A>)
+    fun stop()
+}
+
+class CommonEventStreamReadingStrategy<A : Aggregate>(
+    private val streamManager: EventStreamReaderManager,
+    /**
+     * Allows to map some row representation of event to instance of [Event] class.
+     */
+    private val eventMapper: EventMapper,
+    /**
+     * When we store event to DB we just store the row bytes of the event (most likely json representation).
+     * Also, the event meta-information is stored - id, timestamp, aggregateId and the NAME of the event.
+     * The NAME is mapped to the class of corresponding event. So we ask this function - which class is mapped to the name?
+     * to correctly deserialize the row content of the event.
+     */
+    private val nameToEventClassFunc: (String) -> KClass<Event<A>>,
+    /**
+     * Maps the event classes to corresponding business logic that should be performed once the event of the class is fired.
+     */
+    private val handlers: Map<KClass<out Event<A>>, suspend (Event<A>) -> Unit>,
+) : EventStreamReadingStrategy<A> {
+    private val logger: Logger = LoggerFactory.getLogger(CommonEventStreamReadingStrategy::class.java)
+    private val activityUpdateInterval: Duration = 30.seconds
+
+    @Volatile
+    private var isActive = true
+
+    @Volatile
+    private var activityJob: Job? = null
```

Comment:
In case if subscriber receives no events from stream but is still alive, we should update its activity timestamp to not mislead other subscribers that are ready to intercept long inactive one.